### PR TITLE
browser back button bug fixed

### PR DESCRIPTION
**Pull request by Deepak8717**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description
fixed the bug "wrong route being highlighted when pressed the back button" as  described in issue #284. 

Fixes # (#284 )

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

self tested

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by Deepak8717** [comment]
@atapas @koustov please check

File: src/plays/dynamic-routes/DynamicRoutes.jsx [hunk 1]
```diff
@@ -23,17 +23,23 @@ function DynamicRoutes(props) {
...
   });
-
   const mealType = [];
   data.map((recipe) => {
     return mealType.push(recipe.mealtype); //push  meal categories to an array
   });
 
-  const uniqMealType = [...new Set(mealType)]; // eliminate duplicate categories so we can render a navbar of uniq categories
+  const uniqMealType = [...new Set(mealType)];
+  // eliminate duplicate categories so we can render a navbar of uniq categories
```
Comment (atapas) [inline src/plays/dynamic-routes/DynamicRoutes.jsx:32]: This comment should go above right?

**Comment by Deepak8717** [comment]
@atapas  actually that is for github link of react-play on header. which is same or main github repo link for every project.

**Comment by atapas** [comment]
> @atapas actually that is for github link of react-play on header. which is same or main github repo link for every project. for that we can raise separate issue.

Single line change... related to the same play! Why to create another issue.

got to your play in the play-meta.js file.. change the github link from `https://github.com/reactplay/react-play/tree/main/src/plays/dynamic-routes/:menu` to `https://github.com/reactplay/react-play/tree/main/src/plays/dynamic-routes`... and test it from the preview build.

I hope it helps.

**Comment by atapas** [comment]
> > @atapas actually that is for github link of react-play on header. which is same or main github repo link for every project. for that we can raise separate issue.
> 
> Single line change... related to the same play! Why to create another issue.
> 
> got to your play in the play-meta.js file.. change the github link from `https://github.com/reactplay/react-play/tree/main/src/plays/dynamic-routes/:menu` to `https://github.com/reactplay/react-play/tree/main/src/plays/dynamic-routes`... and test it from the preview build.
> 
> I hope it helps.

@Deepak8717 where are we on this one?

**Comment by Deepak8717** [comment]
@atapas I apologize, for not being able to find any solution.

**Comment by atapas** [comment]
> @atapas I apologize, for not being able to find any solution.

No worries @Deepak8717 , lets get on it together to find a solution.

**Comment by Deepak8717** [comment]
@atapas . i request you to look into this or may be @koustov

**Comment by koustov** [comment]
@Deepak8717 let's do these steps
1. Got to `play-meta.js`
2. Navigate to your play and change [`path`](https://github.com/reactplay/react-play/blob/8537088890a3d510fd3ab126f717dc2d5e74a878/src/meta/play-meta.js#L386) variable to `/plays/dynamic-routes` instead of `/plays/dynamic-routes/:menu` (note: `path` variable is something dynamically created by application, we shouldn't modify it.)
3. Now on `DynamicRoutes.jsx` read the params
  1. If the param is empty or undefined, set the active route to breakfast
  2. Else, set active routes to what you receive from params

This should solve your issue and will automatically fix the broken `github` link

**Comment by koustov** [comment]
> @Deepak8717 let's do these steps
> 
> 1. Got to `play-meta.js`
> 2. Navigate to your play and change [`path`](https://github.com/reactplay/react-play/blob/8537088890a3d510fd3ab126f717dc2d5e74a878/src/meta/play-meta.js#L386) variable to `/plays/dynamic-routes` instead of `/plays/dynamic-routes/:menu` (note: `path` variable is something dynamically created by application, we shouldn't modify it.)
> 3. Now on `DynamicRoutes.jsx` read the params
>    
>    1. If the param is empty or undefined, set the active route to breakfast
>    2. Else, set active routes to what you receive from params
> 
> Note: You can also set optional route params in RouteDefs.jsx This should solve your issue and will automatically fix the broken `github` link

@Deepak8717 you can refer [this sample](https://github.com/reactplay/react-play/tree/dynamic-route-rough-cut)

**Comment by Deepak8717** [comment]
@koustov thank you so much this helped me a lot. @atapas Kindly review

**Review by koustov — CHANGES_REQUESTED** [review]
@Deepak8717 you need to pull master. The code is out of sync. Just check the PR, its showing a lot of unnecessary updates

**Comment by Deepak8717** [comment]
@koustov please check again

**Review by koustov — APPROVED** [review]
lgtm

**Review by atapas — APPROVED** [review]

**Pull request merged**